Cannot compile deal.II 9.1.1 with Trilinos (macOS)

Dear all,

in a [different issue](https://github.com/dealii/dealii/issues/8366) I've found that I cannot compile deal.II with PETSc and Trilinos enabled (macOS 10.14.5 with Apple LLVM version 10.0.1, Trilinos 12.12.1, without Fortran or MPI support, from sources).

This issue is separated from the other as the compilation error is quite different:

```
% make     
[  0%] Built target expand_instantiations_exe
[  0%] Built target doxygen_headers
[  3%] Built target obj_boost_serialization_release
...
[ 55%] Building CXX object source/fe/CMakeFiles/obj_fe_debug.dir/mapping_fe_field_inst2.cc.o
[ 55%] Building CXX object source/fe/CMakeFiles/obj_fe_debug.dir/fe_tools_interpolate.cc.o
In file included from /Users/sensei/Downloads/src/dealii-9.1.1/source/fe/fe_tools_interpolate.cc:17:
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:506:33: error: no member named 'EpetraWrappers' in namespace
      'dealii::LinearAlgebra'
        typename LinearAlgebra::EpetraWrappers::Vector::value_type> &,
                 ~~~~~~~~~~~~~~~^
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:506:57: error: expected a qualified name after 'typename'
        typename LinearAlgebra::EpetraWrappers::Vector::value_type> &,
                                                        ^
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:506:57: error: unknown type name 'value_type'; did you mean
      'IndexSet::value_type'?
        typename LinearAlgebra::EpetraWrappers::Vector::value_type> &,
                                                        ^~~~~~~~~~
                                                        IndexSet::value_type
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/base/index_set.h:104:9: note: 'IndexSet::value_type' declared here
  using value_type = signed int;
        ^
In file included from /Users/sensei/Downloads/src/dealii-9.1.1/source/fe/fe_tools_interpolate.cc:17:
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:507:28: error: no member named 'EpetraWrappers' in namespace
      'dealii::LinearAlgebra'
      const LinearAlgebra::EpetraWrappers::Vector &,
            ~~~~~~~~~~~~~~~^
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:507:44: error: use of class template 'Vector' requires template
      arguments
      const LinearAlgebra::EpetraWrappers::Vector &,
                                           ^
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/lac/vector.h:1043:16: note: template is declared here
  friend class Vector;
               ^
In file included from /Users/sensei/Downloads/src/dealii-9.1.1/source/fe/fe_tools_interpolate.cc:17:
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:510:33: error: no member named 'EpetraWrappers' in namespace
      'dealii::LinearAlgebra'
        typename LinearAlgebra::EpetraWrappers::Vector::value_type> &,
                 ~~~~~~~~~~~~~~~^
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:510:57: error: expected a qualified name after 'typename'
        typename LinearAlgebra::EpetraWrappers::Vector::value_type> &,
                                                        ^
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:510:57: error: unknown type name 'value_type'; did you mean
      'IndexSet::value_type'?
        typename LinearAlgebra::EpetraWrappers::Vector::value_type> &,
                                                        ^~~~~~~~~~
                                                        IndexSet::value_type
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/base/index_set.h:104:9: note: 'IndexSet::value_type' declared here
  using value_type = signed int;
        ^
In file included from /Users/sensei/Downloads/src/dealii-9.1.1/source/fe/fe_tools_interpolate.cc:17:
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:511:22: error: no member named 'EpetraWrappers' in namespace
      'dealii::LinearAlgebra'
      LinearAlgebra::EpetraWrappers::Vector &)
      ~~~~~~~~~~~~~~~^
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/fe/fe_tools_interpolate.templates.h:511:38: error: use of class template 'Vector' requires template
      arguments
      LinearAlgebra::EpetraWrappers::Vector &)
                                     ^
/Users/sensei/Downloads/src/dealii-9.1.1/include/deal.II/lac/vector.h:1043:16: note: template is declared here
  friend class Vector;
               ^
10 errors generated.
make[2]: *** [source/fe/CMakeFiles/obj_fe_debug.dir/fe_tools_interpolate.cc.o] Error 1
make[1]: *** [source/fe/CMakeFiles/obj_fe_debug.dir/all] Error 2
make: *** [all] Error 2
```

I haven't seen any issues regarding Trilinos alone, not at least open.

Am I doing something wrong?

Thanks!
    Franco

--- a/include/deal.II/fe/fe_tools_interpolate.templates.h
+++ b/include/deal.II/fe/fe_tools_interpolate.templates.h
@@ -45,6 +45,7 @@
 #include <deal.II/lac/la_vector.h>
 #include <deal.II/lac/petsc_block_vector.h>
 #include <deal.II/lac/petsc_vector.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

--- a/source/dofs/dof_accessor_get.cc
+++ b/source/dofs/dof_accessor_get.cc
@@ -31,6 +31,7 @@
 #include <deal.II/lac/petsc_block_vector.h>
 #include <deal.II/lac/petsc_vector.h>
 #include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

--- a/source/dofs/dof_accessor_set.cc
+++ b/source/dofs/dof_accessor_set.cc
@@ -31,6 +31,7 @@
 #include <deal.II/lac/petsc_block_vector.h>
 #include <deal.II/lac/petsc_vector.h>
 #include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

--- a/source/fe/fe_values.cc
+++ b/source/fe/fe_values.cc
@@ -38,6 +38,7 @@
 #include <deal.II/lac/la_vector.h>
 #include <deal.II/lac/petsc_block_vector.h>
 #include <deal.II/lac/petsc_vector.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

--- a/source/fe/mapping_fe_field.cc
+++ b/source/fe/mapping_fe_field.cc
@@ -43,6 +43,7 @@
 #include <deal.II/lac/la_vector.h>
 #include <deal.II/lac/petsc_block_vector.h>
 #include <deal.II/lac/petsc_vector.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

--- a/source/lac/solver.cc
+++ b/source/lac/solver.cc
@@ -21,6 +21,7 @@
 #include <deal.II/lac/petsc_block_vector.h>
 #include <deal.II/lac/petsc_vector.h>
 #include <deal.II/lac/solver.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

--- a/source/lac/vector_memory.cc
+++ b/source/lac/vector_memory.cc
@@ -19,6 +19,7 @@
 #include <deal.II/lac/la_vector.h>
 #include <deal.II/lac/petsc_block_vector.h>
 #include <deal.II/lac/petsc_vector.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

--- a/source/multigrid/mg_base.cc
+++ b/source/multigrid/mg_base.cc
@@ -18,6 +18,7 @@
 #include <deal.II/lac/la_parallel_vector.h>
 #include <deal.II/lac/la_vector.h>
 #include <deal.II/lac/petsc_block_vector.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

--- a/source/numerics/derivative_approximation.cc
+++ b/source/numerics/derivative_approximation.cc
@@ -38,6 +38,7 @@
 #include <deal.II/lac/la_vector.h>
 #include <deal.II/lac/petsc_block_vector.h>
 #include <deal.II/lac/petsc_vector.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
 #include <deal.II/lac/trilinos_parallel_block_vector.h>
 #include <deal.II/lac/trilinos_tpetra_vector.h>
 #include <deal.II/lac/trilinos_vector.h>

